README.md

```diff
@@ -65,6 +65,30 @@ object = client.fetch_object_value(flag_key: 'object_value', default_value: JSON
 
 For complete documentation, visit: https://openfeature.dev/docs/category/concepts
 
+### Providers
+
+Providers are the abstraction layer between OpenFeature and different flag management systems. Custom ones can easily be implemented.
+
+The `NoOpProvider` is an example of a minimally functioning provider. For complete documentation on the Provider interface, visit: https://openfeature.dev/specification/sections/providers.
+
+In addition to the `fetch_*` methods, providers can optionally implement lifecycle methods that are invoked when the underlying provider is switched out. For example:
+
+```ruby
+class MyProvider
+  def init
+    # Perform any initialization steps with flag management system here
+    # Return value is ignored
+  end
+
+  def shutdown
+    # Perform any shutdown/reclamation steps with flag management system here
+    # Return value is ignored
+  end
```

Comment:
The spec _tries_ to avoid specifying names to avoid classes with lang keywords, etc.

As long as it's clear and idiomatic, you can use whatever term makes sense.